pkg/asset/manifests/openstack/cloudproviderconfig.go

```diff
@@ -75,3 +71,35 @@ func CloudProviderConfigSecret(cloud *clientconfig.Cloud) ([]byte, error) {
 
 	return []byte(res.String()), nil
 }
+
+func generateCloudProviderConfig(cloudConfig *clientconfig.Cloud) (cloudProviderConfigData, cloudProviderConfigCABundleData string, err error) {
+	cloudProviderConfigData = `[Global]
+secret-name = openstack-credentials
+secret-namespace = kube-system
+`
+	if regionName := cloudConfig.RegionName; regionName != "" {
+		cloudProviderConfigData += "region = " + regionName + "\n"
+	}
+
+	if caCertFile := cloudConfig.CACertFile; caCertFile != "" {
+		cloudProviderConfigData += "ca-file = /etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem\n"
+		caFile, err := ioutil.ReadFile(caCertFile)
+		if err != nil {
+			return "", "", Error{err, "failed to read clouds.yaml ca-cert from disk"}
```

Comment:
Suggestion:

```
return "", "", fmt.Errorf("failed to read clouds.yaml ca-cert from disk: %w", err)
```

I believe this also implements Unwrap?

Reply:
Nice, I had completely forgot about that thing. Thank you! This is definitely going in the next revision

Reply:
Count me as confused. How is using `fmt.Errorf` acceptable when `errors.Wrap` is not? They both satisfy your rationale (1). Neither of them satisfy your rationale (3). And `errors.Wrap` is used extensively throughout the installer codebase, whereas `fmt.Errorf` is not.

Reply:
@mdbooth’s solution has the interesting property of addressing your concern (type maintenance burden), all while not breaking my most cherished point 2 (not an external dependency). I can give up 3 if nobody cares about the “sentinel” pattern :)